az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -376,9 +378,57 @@ public static void proxyUserKillAllSpawnedHadoopJobs(
         final UserGroupInformation proxyUser =
             HadoopSecureWrapperUtils.setupProxyUserWithHSM(hadoopSecurityManager, properties,
                 tokenFile.getAbsolutePath(), log);
+
+        // todo: print tokens in proxyUser
+        log.debug("proxyUserKillAllSpawnedHadoopJobs.proxyUser = " + proxyUser);
+        for (Token<?> token : proxyUser.getCredentials().getAllTokens()) {
```

Comment:
Is proxyUser.getCredentials() alway non-null, or is there a chance of NPE here ?

Reply:
it's always non-null as we observed